hibernate-core/src/main/java/org/hibernate/query/internal/NamedObjectRepositoryImpl.java

```diff
@@ -36,22 +37,44 @@
 	private static final Logger log = Logger.getLogger( NamedObjectRepository.class );
 
 	private final Map<String, NamedHqlQueryMemento> hqlMementoMap;
+	private final Map<String, NamedSqmQueryMemento> sqmMementoMap;
```

Comment:
IMO it would be better to rename the `NamedHqlQueryMemento` to `NamedSqmQueryMemento` and only have a single map.

Reply:
not sure about this, the two implementations do slightly different things. The code seems a little more readable with the two impl

Reply:
You can have two implementations, but the interface can be the same and hence you only need a single map and registration code.

Reply:
what do you think of https://github.com/hibernate/hibernate-orm/pull/4782

Reply:
IMO the new PR is much better. Want to close this one now?